app/notifications/process_notifications.py

```diff
@@ -222,13 +227,15 @@ def send_to_queue_for_recipient_info_based_on_recipient_identifier(
     else:
         tasks = [
             lookup_va_profile_id.si(notification.id).set(queue=QueueNames.LOOKUP_VA_PROFILE_ID),
+            send_va_onsite_notification_task.si(notification.template.id, onsite_enabled)
+                                            .set(queue=QueueNames.SEND_ONSITE_NOTIFICATION),
             lookup_contact_info.si(notification.id).set(queue=QueueNames.LOOKUP_CONTACT_INFO),
             deliver_task.si(notification.id).set(queue=deliver_queue)
         ]
 
         if is_feature_enabled(FeatureFlag.CHECK_RECIPIENT_COMMUNICATION_PERMISSIONS_ENABLED) and communication_item_id:
             tasks.insert(
-                2,
+                len(tasks) - 1,
```

Comment:
Same as above. I believe this should be an `append` instead of `insert` unless I have a misunderstanding as to the goal.

Reply:
Same situation. It inserts *before* the last element. I changed it to hopefully be more robust, since I'm adding something into the queue that wasn't there before, now it will respond to the size appropriately.